src/screens/DiscoveryScreenV2/elements/DiscoveryTab.js

```diff
@@ -1,44 +1,64 @@
 import * as React from 'react';
 import {Dimensions, Keyboard, Pressable, ScrollView, StyleSheet, Text, View} from 'react-native';
 
+import {useRoute} from '@react-navigation/core';
+import PropTypes from 'prop-types';
 import {colors} from '../../../utils/colors';
-import {fonts, normalizeFontSize} from '../../../utils/fonts';
+import {normalizeFontSize} from '../../../utils/fonts';
+import {setNavbarTitle} from '../../../context/actions/setMyProfileAction';
+import {Context} from '../../../context';
 
-const {width} = Dimensions.get('screen');
+const windowWidth = Dimensions.get('window').width;
+
+const DiscoveryTab = ({onChangeScreen, selectedScreen = 0, tabs}) => {
+  const [, dispatchNavbar] = React.useContext(Context).profile;
+  const route = useRoute();
 
-const DiscoveryTab = ({onChangeScreen, selectedScreen = 0}) => {
   const handleTabOnClicked = React.useCallback((index) => {
     Keyboard.dismiss();
     onChangeScreen(index);
   }, []);
 
-  const tabs = ['Users', 'Communities', 'Domains', 'News'];
+  const handleChangeTitle = (index = 0) => {
+    const title = ['Search Users', 'Your Communities', 'Your Domains'][index] || '';
+    setNavbarTitle(title, dispatchNavbar);
+  };
+
+  React.useEffect(() => {
+    if (route.name === 'Followings') handleChangeTitle();
+  }, []);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The component now uses context and route information to modify its behavior. This is a significant change and should be thoroughly tested to ensure it doesn't introduce any regressions or unexpected behavior. The useEffect hook at line 27-29 seems to only run once due to an empty dependency array. If the intention was to run this effect whenever the route name changes, you should add `route.name` to the dependency array.
```diff
- const tabs = ['Users', 'Communities', 'Domains', 'News'];
+ const [, dispatchNavbar] = React.useContext(Context).profile;
+ const route = useRoute();
+ 
+ const handleChangeTitle = (index = 0) => {
+   const title = ['Search Users', 'Your Communities', 'Your Domains'][index] || '';
+   setNavbarTitle(title, dispatchNavbar);
+ };
+ 
+ React.useEffect(() => {
+   if (route.name === 'Followings') handleChangeTitle();
+ }, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->